File: 2020-javascript/Day 9/advent-9-1.py
numbers = []
with open("Day 9/advent-9-input.txt", "r") as data:
  lines = data.readlines()
  numbers = [int(num.rstrip("\n")) for num in lines]

preamble = numbers[0:25]
numbers = numbers[25:]
target = numbers[0]

def brute_forcer(preamble, target):
  found = False
  for i in range(0, 25):
    num1 = preamble[i];
    for j in range(0, 25):
      if i == j:
        continue
      num2 = preamble[j]
      if num1 + num2 == target:
        found = True
        break
    if found:
      return True
  if found == False:
    print(target)
    return False

while brute_forcer(preamble, target):
  preamble.pop(0)
  preamble.append(target)
  numbers.pop(0)
  target = numbers[0]
  if target is None:
    print("Null target!")
    break

File: 2020-javascript/Day 9/advent-9-1.js
const fs = require("fs");

function readData(err, data) {
  if (err) {
    console.log("Error!" + "\n" + err.message);
  } else {
    const numbers = parseData(data);
    const preamble = numbers.splice(0, 25);
    let target = numbers[0];
    
    while (bruteForcer(preamble, target)) {
      preamble.shift();
      preamble.push(target);
      numbers.shift();
      target = numbers[0];
      if (target === undefined) break;
    }
    console.log("WHILE LOOP OVER")
  }
}

// Returns true if target is found from the preamble numbers
function bruteForcer(preamble, target) {
  let found = false;
  for (let i = 0; i < preamble.length; i++) {
    const firstNum = preamble[i];
    for (let j = 0; j < preamble.length; j++) {
      if (i === j) continue;
      const secondNum = preamble[j];
      if (firstNum + secondNum === target) {
        found = true;
        break;
      }
    }
    if (found === true) return true;
  }
  if (found === false) {
    console.log(target);
    return false;
  }
}

// Parse Data
function parseData(data) {
  splitData = data.split("\n");
  numbers = splitData.map(stringNum => Number(stringNum));
  return numbers;
}

fs.readFile("Day\ 9/advent-9-input.txt", "UTF-8", readData);